src/nominatim_api/results.py

```diff
@@ -225,23 +208,40 @@ def lon(self) -> float:
         """
         return self.centroid[0]
 
+    @property
+    def display_name(self) -> Optional[str]:
+        """ Dynamically compute the display name for the result place
+            and, if available, its address information..
+        """
+        # No longer sets the locale name, want to do that explicitly in format.py
+        # using the formatter object
+        if self.address_rows:  # if this is true we need additional processing
+            label_parts: List[str] = []
+
+            for line in self.address_rows:  # assume locale_name is set by external formatter
+                if line.isaddress and line.names and line.local_name:  # checks if it is an address
+                    if not label_parts or label_parts[-1] != line.local_name:
+                        label_parts.append(line.local_name)
```

Comment:
You need to have a fallback here as well. It is probably easiest to add a `display_name` property to AddressLine as well which returns the first of (locale_name, names['name'], next(iter(self.names.values()), None).